pkg/client/registration.go

```diff
@@ -14,33 +15,44 @@ import (
 // as a keep member candidate.
 // Also, once the client is confirmed as registered, it triggers the monitoring
 // process to keep the operator's status up to date in the pool.
+// If operator status in the pool cannot be monitored, e.g. when operator is
+// removed from the pool it triggers the registration process from the begining.
 func checkStatusAndRegisterForApplication(
 	ctx context.Context,
 	ethereumChain eth.Handle,
 	application common.Address,
 ) {
-	isRegistered, err := ethereumChain.IsRegisteredForApplication(application)
-	if err != nil {
-		logger.Errorf(
-			"failed to check if member is registered for application [%s]: [%v]",
-			application.String(),
-			err,
-		)
-		return
-	}
+Registration:
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			isRegistered, err := ethereumChain.IsRegisteredForApplication(application)
+			if err != nil {
+				logger.Errorf(
+					"failed to check if member is registered for application [%s]: [%v]",
+					application.String(),
+					err,
+				)
+				return
```

Comment:
Should we return? Since we are in the registration loop maybe it makes sense to retry in the next iteration?